packages/plugin-ext/src/hosted/node/hosted-instance-manager.ts

```diff
@@ -232,7 +233,21 @@ export abstract class AbstractHostedInstanceManager implements HostedInstanceMan
     }
 
     protected async getStartCommand(port?: number, debugConfig?: DebugConfiguration): Promise<string[]> {
-        const command = ['yarn', 'theia', 'start'];
+
+        const processArguments = process.argv;
+        let command: string[];
+        if (environment.electron.is()) {
+            command = ['yarn', 'theia', 'start'];
+        } else {
+            command = processArguments.filter(arg => {
+                // remove --port= argument if set
```

Comment:
What if main Theia is started on default port?
In this case we will end up with an attempt to start hosted instance on the same port as main Theia. Or did I miss something?

Reply:
if started on default port, then there is no --port in original process and at line https://github.com/theia-ide/theia/blob/eb528a71d0b54b69b50f3d0e2feb5f621ae02076/packages/plugin-ext/src/hosted/node/hosted-instance-manager.ts#L254 it will add `--port=3030`

Reply:
Still cannot get it completely. Hosted instance manager does staff only for hosted instance and has nothing in common to start main Theia. What the reason to remove port here then (if it is hosted instance related)?

Reply:
@mmorhun then what does 'yarn theia start' command ? ( what is done right now ?)

Reply:
Got it, thanks, it just reuses the command with which main Theia was started